src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -76,10 +77,7 @@ private void checkIsAtLeastEditorInCourseElseThrow(@NotNull Course course, @Null
      * @return true if the passed user is at least an editor in the course, false otherwise
      */
     public boolean isAtLeastEditorInCourse(@NotNull Course course, @Nullable User user) {
-        if (user == null || user.getGroups() == null) {
-            // only retrieve the user and the groups if the user is null or the groups are missing (to save performance)
-            user = userRepository.getUserWithGroupsAndAuthorities();
-        }
+        user = loadUserIfNeeded(user);
```

Comment:
If someone has a suggestion on how to write this nicely without a warning, feel free to add your suggestions.
Simply adding a new variable does not really seam helpful to me in this case.

Reply:
I think we can ignore the suggestion in this case, as in the current implementation the callers depend on it that the user parameter gets replaced with the user with groups.